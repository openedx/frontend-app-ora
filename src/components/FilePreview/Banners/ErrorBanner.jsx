import React from 'react';
import PropTypes from 'prop-types';

import { Alert, Button } from '@edx/paragon';
import { Info } from '@edx/paragon/icons';
import { useIntl } from '@edx/frontend-platform/i18n';

const messageShape = PropTypes.shape({
  id: PropTypes.string,
  defaultMessage: PropTypes.string,
});

export const ErrorBanner = ({ actions, headerMessage, children }) => {
  const { formatMessage } = useIntl();
  const actionButtons = actions.map(action => (
    <Button key={action.id} onClick={action.onClick} variant="outline-primary">
      {formatMessage(action.message)}
    </Button>
  ));

  return (
    <Alert variant="danger" icon={Info} actions={actionButtons}>
      <Alert.Heading>
        {formatMessage(headerMessage)}
      </Alert.Heading>
      {children}
    </Alert>
  );
};
ErrorBanner.defaultProps = {
  actions: [],
  children: null,
};
ErrorBanner.propTypes = {
  actions: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      onClick: PropTypes.func,
      message: messageShape,
    }),
  ),
  headerMessage: messageShape.isRequired,
  children: PropTypes.node,
};

export default ErrorBanner;
