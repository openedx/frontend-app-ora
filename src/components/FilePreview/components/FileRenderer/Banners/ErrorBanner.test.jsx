import { shallow } from '@edx/react-unit-test-utils';

import ErrorBanner from './ErrorBanner';

describe('<ErrorBanner />', () => {
  const props = {
    headerMessage: {
      id: 'headerMessageId',
      defaultMessage: 'headerMessage',
    },
    actions: [
      {
        id: 'actionId',
        onClick: jest.fn().mockName('onClick'),
        message: {
          id: 'actionMessageId',
          defaultMessage: 'actionMessage',
        },
      },
    ],
  };

  it('renders correctly', () => {
    const wrapper = shallow(<ErrorBanner {...props}>children</ErrorBanner>);
    expect(wrapper.snapshot).toMatchSnapshot();
  });

  it('renders without actions', () => {
    const wrapper = shallow(<ErrorBanner headerMessage={props.headerMessage}>children</ErrorBanner>);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
