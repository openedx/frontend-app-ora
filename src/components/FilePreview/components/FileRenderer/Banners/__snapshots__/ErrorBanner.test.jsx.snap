// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBanner /> renders correctly 1`] = `
<Alert
  actions={
    Array [
      <Button
        onClick={[MockFunction onClick]}
        variant="outline-primary"
      >
        actionMessage
      </Button>,
    ]
  }
  variant="danger"
>
  <Alert.Heading>
    headerMessage
  </Alert.Heading>
  children
</Alert>
`;

exports[`<ErrorBanner /> renders without actions 1`] = `
<Alert
  actions={Array []}
  variant="danger"
>
  <Alert.Heading>
    headerMessage
  </Alert.Heading>
  children
</Alert>
`;
